File: OOPS_Concepts/Java/Inheritance.java
/**
 * Inheritance in Java
 * 
 * Inheritance is a mechanism in which one class acquires the property of another class.
 * The class which inherits the properties of another is known as the subclass (derived class, child class),
 * and the class whose properties are inherited is known as the superclass (base class, parent class).
 */

class Vehicle {
    String make;
    String model;

    void startEngine() {
        System.out.println("Engine started.");
    }
}

class Car extends Vehicle {
    int numberOfDoors;

    void honk() {
        System.out.println("Car honking.");
    }
}

public class Main {
    public static void main(String[] args) {
        Car car = new Car();
        car.make = "Toyota";
        car.model = "Camry";
        car.numberOfDoors = 4;

        car.startEngine();
        car.honk();

        System.out.println("Make: " + car.make + ", Model: " + car.model + ", Number of Doors: " + car.numberOfDoors);
    }
}

File: OOPS_Concepts/CSharp/Inheritance.cs
using System;

public class Vehicle
{
    public string Make { get; set; }
    public string Model { get; set; }

    public void StartEngine()
    {
        Console.WriteLine("Engine started.");
    }
}

public class Car : Vehicle
{
    public int NumberOfDoors { get; set; }

    public void Honk()
    {
        Console.WriteLine("Car honking.");
    }
}

public class Program
{
    public static void Main()
    {
        Car car = new Car();
        car.Make = "Toyota";
        car.Model = "Camry";
        car.NumberOfDoors = 4;

        car.StartEngine();
        car.Honk();

        Console.WriteLine($"Make: {car.Make}, Model: {car.Model}, Number of Doors: {car.NumberOfDoors}");
    }
}
